src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml.cs

```diff
@@ -113,8 +120,55 @@ public async Task<IActionResult> OnGetAsync(string id, string revisionId = null)
             return Page();
         }
 
+        private InlineDiffLine<CodeLine>[] CreateDiffOnlyLines(InlineDiffLine<CodeLine>[] lines)
+        {
+            var unchangedQueue = new Queue<InlineDiffLine<CodeLine>>();
+            var filteredLines = new List<InlineDiffLine<CodeLine>>();
+            for (int i = 0; i < lines.Count(); i++)
+            {
+                if (lines[i].Kind == DiffLineKind.Unchanged)
+                {
+                    // Hold last two unchanged line in queue to create context
+                    unchangedQueue.Enqueue(lines[i]);
+                    if (unchangedQueue.Count > REVIEW_DIFF_CONTEXT_SIZE)
+                    {
+                        unchangedQueue.Dequeue();
+                    }
+                }
+                else
+                {
+                    if (unchangedQueue.Count > 0)
+                    {
+                        // Add sepearator to show skipping lines. for e.g. .....
+                        if (filteredLines.Count > 0)
+                        {
+                            filteredLines.Add(new InlineDiffLine<CodeLine>(new CodeLine(DIFF_CONTEXT_SEPERATOR, null), DiffLineKind.Unchanged));
+                        }
+                        // Copy pre context
+                        filteredLines.AddRange(unchangedQueue);
+                        unchangedQueue.Clear();
+                    }
+                    filteredLines.Add(lines[i]);
+                    // Add post context
+                    int contextIndex = i + 1, contextEnd = i + REVIEW_DIFF_CONTEXT_SIZE;
+                    while (contextIndex <= contextEnd && contextIndex < lines.Count() && lines[contextIndex].Kind == DiffLineKind.Unchanged)
+                    {
+                        filteredLines.Add(lines[contextIndex]);
```

Comment:
You are using indexing into lines to add post context but a queue to collect pre-context. Can we do the pre-context the same way?

Reply:
Reason for using queue is to avoid more complex logic to correctly find pre context which is not duplicate if diff happened to be within small range itself. Actually I think I can try to keep track of the index added to avoid duplicate and get rid of this queue based implementation.